h2o-py/tests/testdir_sklearn/pyunit_sklearn_generic_all_estimators.py

```diff
@@ -0,0 +1,208 @@
+
+        if _get_custom_behaviour(estimator_cls).get('predict_proba', True):
+            probs = estimator.predict_proba(data.X_test)
+            print(probs)
+            assert probs.shape == (len(data.X_test), 2)
+            assert np.allclose(np.sum(probs, axis=1), 1.), "`predict_proba` didn't return probabilities"
+        else:
+            try:
+                estimator.predict_proba(data.X_test)
+            except AttributeError as e:
+                assert "No `predict_proba` method" in str(e)
+
+        if _get_custom_behaviour(estimator_cls).get('score', True):
+            score = estimator.score(data.X_test, data.y_test)
+            assert isinstance(score, float)
+            skl_score = accuracy_score(data.y_test, preds)
+            assert abs(score - skl_score) < 1e-6
+            scores[estimator_cls].update(with_numpy_arrays=score)
+
+
+def test_scores_are_equivalent(estimator_cls):
+    try:
+        lk, rk = ('with_h2o_frames', 'with_numpy_arrays')
+        est_scores = scores[estimator_cls]
+        if lk in est_scores and rk in est_scores:
+            assert abs(est_scores[lk] - abs(est_scores[rk])) < 1e-6, \
+                "expected equivalent scores but got {lk}={lscore} and {rk}={rscore}" \
+                    .format(lk=lk, rk=rk, lscore=est_scores[lk], rscore=est_scores[rk])
+        elif lk not in est_scores:
+            print("no scores for {}".format(estimator_cls.__name__+' '+lk))
+        else:
+            print("no scores for {}".format(estimator_cls.__name__+' '+rk))
+    except AssertionError as e:
+        if _get_custom_behaviour(estimator_cls).get('scores_may_differ', False):
+            print("ERROR !!! "+str(e))
+        else:
+            raise e
+
+
+def make_test(test, classifier):
+    bound_test = partial(test, classifier)
+    bound_test.__name__ = test.__name__
+    pyunit_utils.tag_test(bound_test, classifier.__name__)
+    return bound_test
+
+
+def make_tests(classifier):
+    return map(lambda test: make_test(test, classifier), [
+        test_estimator_with_h2o_frames,
+        test_estimator_with_numpy_arrays,
+        test_scores_are_equivalent
+    ])
+
+
+failing = [
+    'H2OAggregatorEstimator',  # can't use predict with it: looks more like a transformer, maybe should be removed from sklearn API
+    'H2OCoxProportionalHazardsEstimator',  # NPE at water.fvec.Frame.<init>(Frame.java:168) .... at hex.coxph.CoxPH$CollectTimes.collect(CoxPH.java:805)
+    'H2ODeepWaterEstimator',  # requires DW backend
+    'H2OGenericEstimator',  # maybe should be removed from sklearn API
+    'H2OStackedEnsembleEstimator',  # needs a separate test (requires models as parameters)
+    'H2OSingularValueDecompositionEstimator', # H2OResponseError: Server error water.exceptions.H2ONotFoundArgumentException: Error: POST /3/ModelBuilders/svd not found
+    'H2OWord2vecEstimator',  # needs a separate test (requires pre_trained model as parameter)
+]
```

Comment:
reviewers, please confirm if this is normal...

Reply:
SVD error is fixed [here](https://github.com/h2oai/h2o-3/pull/3757), wrong api version was called.

Reply:
Reenabled it, seems to work now, thanks.